lib/Cake/Routing/Router.php

```diff
@@ -619,8 +619,12 @@ public static function parse($url) {
 			$out['action'] = $out['prefix'] . '_' . $out['action'];
 		}
 
-		if (!empty($ext) && !isset($out['ext'])) {
-			$out['ext'] = $ext;
+		if (!empty($ext)) {
+			if (!isset($out['ext'])) {
+				$out['ext'] = $ext;
+			} elseif (is_array($out['ext'])) {
+				$out['ext'] = $ext;
```

Comment:
Would it make sense to normalize and *always* use an array? e.g. `$out['ext'] = (array) $ext` (Some tests will have to be modified for that)